internal/work/dispatcher.go

```diff
@@ -340,3 +321,42 @@ func (d *Dispatcher) senderName(sender dbus.Sender) (string, error) {
 
 	return "", fmt.Errorf("cannot get name for sender: %v", sender)
 }
+
+// workerEventFromSignal creates an ipc.WorkerEvent from a DBus signal.
+func workerEventFromSignal(s *dbus.Signal) (*ipc.WorkerEvent, error) {
+	if s.Name != "com.redhat.Yggdrasil1.Worker1.Event" {
+		return nil, errIncorrectSignal
+	}
+
+	if len(s.Body) < 2 {
+		return nil, errMissingRequiredParameters
```

Comment:
We could improve error message here and tell user/developer what parameters are required.